tools/pipelines/templates/include-setup-npmrc-for-download.yml

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+# Licensed under the MIT License.
+
+# The include-setup-npmrc-for-download template creates an .npmrc file which uses the appropriate feeds from our ADO
+# org to download packages from the right locations, adds the necessary authentication to it, and sets configuration so
+# for the rest of the job execution all invocations of npm (EXCEPT installs through Npm@ tasks) use it.
+# The file is created in a "global" location that should not interfere with anything else.
+#
+# Npm@ tasks always override the user-level .npmrc file with one of their own, so they do not work with this
+# configuration. For that reason, using them to install dependencies should be avoided. Using them for other kinds
+# of npm invocations should be fine.
+
+parameters:
+# Location for the .npmrc file.
+# NOTE: Do not override the default value. It's a parameter just for reusability.
+- name: npmrcLocation
+  type: string
+  default: $(Agent.TempDirectory)/global-download-npmrc
+
+steps:
+# Force a parsing error so the pipeline run cannot even start if the conditions are incorrect for the use of this template.
+# Note this will always show up as a syntax error/warning (squiggly line) in IDEs that validate the structure of the
+# pipeline YAML file.
+- ${{ if ne(variables['System.TeamProject'], 'internal')}}:
+  - "Something tried to use the 'include-setup-npmrc-for-download.yml' template outside the 'internal' ADO project. This is not supported."
```

Comment:
This results in errors like this if the template is incorrectly used under the wrong conditions:
![image](https://github.com/user-attachments/assets/ad4a73ae-db93-40b7-9e7a-13e9e6541de9)